### PR TITLE
8344231: SecurityManager cleanup in java.lang.module and jdk.internal.module

### PR DESCRIPTION
**Pull request by eirbjo**
Please review this PR which removes code depending on SecurityManager from the `java.lang.module` and `jdk.internal.module` packages:

* `ModuleFinder::ofSystem` is updated to not check the `"accessSystemModules"` RuntimePermission
* The `ModuleReferences.ExplodedModuleReader` constructor is updated to remove the check that the caller has access to the directory.
* `Modules::addProvides` is updated to not use `AccessController::doPrivileged` when getting the module class loader
* `SystemModuleFinder::ofSystem` is updated to not returned a wrapped result calling `AccessController::doPrivileged`
* `SystemModuleFinders.SystemImage` is updated to remove a class level comment referring to the SM
* `SystemModuleFinders.SystemModuleReader::checkPermissionToConnect` is removed. The URI constructor parameter in `SystemModuleReader` becomes unused and is removed.

Testing: I have run `test/jdk/java/lang/module`, `test/jdk/java/lang/ModuleTests`,  and `test/jdk/java/lang/ModuleLayer` successfully. GHA results pending.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344231](https://bugs.openjdk.org/browse/JDK-8344231): SecurityManager cleanup in java.lang.module and jdk.internal.module (**Enhancement** - P4)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22113/head:pull/22113` \
`$ git checkout pull/22113`

Update a local copy of the PR: \
`$ git checkout pull/22113` \
`$ git pull https://git.openjdk.org/jdk.git pull/22113/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22113`

View PR using the GUI difftool: \
`$ git pr show -t 22113`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22113.diff">https://git.openjdk.org/jdk/pull/22113.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22113#issuecomment-2477355910)
</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/module/Modules.java [hunk 1]
```diff
@@ -155,9 +153,7 @@ public static void addUses(Module m, Class<?> service) {
     public static void addProvides(Module m, Class<?> service, Class<?> impl) {
         ModuleLayer layer = m.getLayer();
 
-        PrivilegedAction<ClassLoader> pa = m::getClassLoader;
-        @SuppressWarnings("removal")
-        ClassLoader loader = AccessController.doPrivileged(pa);
+        ClassLoader loader = m.getClassLoader();
 
         ClassLoader platformClassLoader = ClassLoaders.platformClassLoader();
```
Comment (AlanBateman) [inline src/java.base/share/classes/jdk/internal/module/Modules.java:158]: You can remove the blank line at L157 now, it was only there to make the doPriv use clearer.

File: src/java.base/share/classes/jdk/internal/module/SystemModuleFinders.java [hunk 2]
```diff
@@ -378,8 +362,6 @@ public byte[] generate(String algorithm) {
 
     /**
      * Holder class for the ImageReader
```
Comment (AlanBateman) [inline src/java.base/share/classes/jdk/internal/module/SystemModuleFinders.java:364]: In passing, you would mind adding a full stop after ImageReader. Only noticed it with the removal of the apinote.

**Review by AlanBateman — APPROVED** [review]
Looks good, thanks for taking this one.

File: src/java.base/share/classes/jdk/internal/module/Modules.java (hunk 1 above)
Comment (eirbjo) [reply src/java.base/share/classes/jdk/internal/module/Modules.java:157]: Removed blank line

File: src/java.base/share/classes/jdk/internal/module/SystemModuleFinders.java (hunk 2 above)
Comment (eirbjo) [reply src/java.base/share/classes/jdk/internal/module/SystemModuleFinders.java]: Added a full stop at the end of the SystemImage class description.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**